### ISSUE TITLE
Support for enabling/disabling hostNetwoking

### ISSUE THREAD

**Issue by prasus** [scylladb/scylla-operator#520]
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**

The operator doesn't support changing hostNetworking yet, It would be great if we can add support for this as we need to disable `hostNetworking` in some use cases. for instance, linkerd doesn't work well with `hostNetworking` as explained [here](https://github.com/linkerd/linkerd2/issues/5938)

The workaround option is to edit each and every statefulset objects one by one and then remove `hostNetworking: true` flag, this method works well, but would be nice if this can be directly controlled by the CRD 🙂 

Thanks! 🙏🏿

**Comment by tnozicka** [comment]
You are right, we one planing to enhance the reconcilation in one of the next releases.
